### PR TITLE
Use toolchain from environment variables

### PR DESCRIPTION
**Pull request by adrian-friedli**
We are using Poky Linux for our embedded devices. Poky sets the usual environment variables for selecting the toolchain. In this pull request I stop overriding already set environment variables and only set a default value where nothing is provided. This makes the build simpler.

### CODE REVIEW

File: Makefile.inc [hunk 1]
```diff
@@ -13,14 +13,15 @@
 # NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE
 # USE OR PERFORMANCE OF THIS SOFTWARE.
 #
-CC = gcc
+CC ?= gcc
```
Comment (apnavik) [inline Makefile.inc]: You can remove it. 
It allows to take default host tool-chain and compatible with cross-compiler as well.

File: Makefile.inc [hunk 2]
```diff
@@ -13,14 +13,15 @@
...
+AR ?= ar
 LIB=lib
 DUT=dut
 CA=ca
 UCC=ucc
 CON=console_src
 WTG=WTGService
-MAKE=make
+MAKE?=make
```
Comment (apnavik) [inline Makefile.inc]: same as above.

File: WTGService/Makefile [hunk 3]
```diff
@@ -13,7 +13,7 @@
 # NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE
 # USE OR PERFORMANCE OF THIS SOFTWARE.
 #
-CC=gcc
+CC?=gcc
```
Comment (apnavik) [inline WTGService/Makefile]: you can remove it.

File: console_src/Makefile [hunk 4]
```diff
@@ -14,7 +14,7 @@
 # USE OR PERFORMANCE OF THIS SOFTWARE.
 #
 
-CC	= gcc
+CC	?= gcc
 CFLAGS	= -g -DWFA_DEBUG -pthread
```
Comment (apnavik) [inline console_src/Makefile]: same as above.

**Comment by adrian-friedli** [comment]
@apnavik Thanks for the review. I removed the lines you stated and I also removed the line for AR, because it also has a default value in make.

**Comment by rettichschnidi** [comment]
Will this ever get merged?

**Pull request closed without merging**